app/controllers/search_controller.rb

```diff
@@ -63,6 +63,19 @@ def chat_channels
     render json: { result: ccm_docs }
   end
 
+  def chat_channels_discoverable
+    ccm_docs_original = Search::ChatChannelMembership.search_documents(
+      params: chat_channel_params.merge(user_id: current_user.id).to_h,
+    )
+    discoverable_params = chat_channel_params.merge(user_id: "all").to_h
+
+    ccm_docs_discoverable = Search::ChatChannelMembership.search_documents(
```

Comment:
If you update the docs and search to use the `viewable_by` field as [suggested here](https://github.com/thepracticaldev/dev.to/pull/7385#discussion_r411386390) then you won't need to make 2 requests to Elasticsearch.

Reply:
Ok @mstruve, we're looking more to optimize this piece of code as you've suggested.
Thanks